### PR TITLE
Avatar img heading

### PR DESCRIPTION
**Pull request by HerrTopi**

### CODE REVIEW

File: packages/ui-img/src/Img/index.js [hunk 1]
```diff
@@ -21,213 +21,156 @@
...
+        {
+          !hasBackground && <img {...imageProps} {...a11yProps} /> // eslint-disable-line jsx-a11y/alt-text
+        }
+        {overlay && (
+          <span
+            css={styles.overlay}
+            style={{
+              backgroundColor: overlay.color,
+              opacity: overlay.opacity * 0.1,
+              mixBlendMode: overlay.blend ? overlay.blend : null
```
Comment (Brailor) [inline packages/ui-img/src/Img/index.js:114]: Shouldn't we move this styling logic to the `generateStyle` function?

File: packages/ui-img/src/Img/index.js [hunk 2]
```diff
@@ -21,213 +21,156 @@
...
-      )
-    }
+    return (
+      <View
+        {...rootProps}
+        as="span"
+        css={styles.view}
+        style={{
+          backgroundImage: hasBackground ? `url(${src})` : undefined
+        }}
```
Comment (Brailor) [inline packages/ui-img/src/Img/index.js:103]: Shouldn't we move this styling logic to the `generateStyle` function?

File: packages/ui-spinner/src/Spinner/index.js [hunk 3]
```diff
@@ -22,7 +22,7 @@
  * SOFTWARE.
  */
 /** @jsx jsx */
-import React, { useMemo } from 'react'
+import { useMemo } from 'react'
```
Comment (Brailor) [inline packages/ui-spinner/src/Spinner/index.js:25]: Does the component render correctly even without `React` being in context?

File: packages/ui-heading/src/Heading/styles.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  const colorStyles = {
+    inherit: { color: 'inherit' },
+    primary: { color: componentTheme.primaryColor },
+    secondary: { color: componentTheme.secondaryColor },
+    'primary-inverse': { color: componentTheme.primaryInverseColor },
+    'secondary-inverse': { color: componentTheme.secondaryInverseColor }
+  }
+  const ellipsisStyle = ellipsis
```
Comment (ke1k0) [inline packages/ui-heading/src/Heading/styles.js:82]: I'd prefer having empty line after blocks.

File: packages/ui-heading/src/Heading/styles.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    top: {
+      paddingTop: componentTheme.borderPadding,
+      borderTop: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`
+    },
+    bottom: {
+      paddingBottom: componentTheme.borderPadding,
+      borderBottom: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`
+    }
+  }
+  return {
```
Comment (ke1k0) [inline packages/ui-heading/src/Heading/styles.js:101]: I'd like to have empty lines before return too.

File: packages/ui-img/src/Img/__tests__/Img.test.js [hunk 6]
```diff
@@ -60,106 +58,6 @@ describe('<Img />', () => {
 
   it('should render an overlay color', async () => {
     await mount(<Img src={image} overlay={{ color: '#ff0000', opacity: 7 }} />)
-    expect(await ImgLocator.find(`.${styles.overlay}`)).to.exist()
-  })
-
-  it('should render a blur filter', async () => {
```
Comment (ke1k0) [inline packages/ui-img/src/Img/__tests__/Img.test.js:66]: Why have these been deleted? Not possible to do with emotion?

File: packages/ui-img/src/Img/index.js [hunk 7]
```diff
@@ -21,213 +21,156 @@
...
+  const a11yProps = {
+    alt: alt || ''
   }
 
-  get supportsObjectFit() {
-    return supportsObjectFit()
-  }
+  // if browser does not support ObjectFit CSS, and Img needs "constrain",
+  // serve up a background-image instead
+  const hasBackground = !supportsObjectFit() && props.constrain
```
Comment (ke1k0) [inline packages/ui-img/src/Img/index.js:63]: style logic like these could be moved to styles.js, we can use props/state/imports there as well

File: packages/ui-img/src/Img/index.js [hunk 8]
```diff
@@ -21,213 +21,156 @@
...
-    } else if (this.props.withGrayscale || this.props.grayscale) {
-      return grayscale
-    } else if (this.props.withBlur || this.props.blur) {
-      return blur
-    } else {
-      return null
-    }
+  const imageProps = {
+    css: styles.img,
+    style: {
```
Comment (ke1k0) [inline packages/ui-img/src/Img/index.js:72]: these could move to styles.js too

File: packages/ui-img/src/Img/styles.js [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+    constrain === 'cover' ? containerCover : containerContain
+
+  const containerBackgroundStyle = {
+    backgroundRepeat: 'no-repeat',
+    backgroundPosition: 'center',
+    backgroundSize: constrain
+  }
+
+  const viewBackgroundImageStyle = hasBackground ? `url(${src})` : {}
+  return {
```
Comment (ke1k0) [inline packages/ui-img/src/Img/styles.js:104]: please add the linebreaks before return statements and after blocks :)

File: packages/ui-docs-client/src/KitchenSink/index.js [hunk 10]
```diff
@@ -24,31 +24,32 @@
...
-    <Avatar
-      key={'avatar4'}
-      name="Rarah Sobinson"
-      size="x-large"
-      src={
-        'https://thumbs.dreamstime.com/b/cute-smiling-teacher-tutor-avatar-internet-learning-line-distant-education-web-study-school-chat-webinar-discussion-forum-185531323.jpg'
-      }
-    />
-  ],
-  Pill: [<Pill key={'pill1'}>Pill teszt 1</Pill>],
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-docs-client/src/KitchenSink/index.js:51]: Not working as far as I know. We should only present working components here

File: packages/ui-spinner/src/Spinner/index.js (hunk 3 above)
Comment (HerrTopi) [reply packages/ui-spinner/src/Spinner/index.js:25]: It makes no visible difference and the linter thinks it's unnecessary

File: packages/ui-docs-client/src/KitchenSink/index.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (ke1k0) [reply packages/ui-docs-client/src/KitchenSink/index.js:51]: Hm, it used to work, but I can see now, it is breaking after the View component refact. Will look into it.

File: packages/ui-heading/src/Heading/index.js [hunk 11]
```diff
@@ -21,131 +21,126 @@
...
 
-    /* eslint-disable react/require-default-props */
-    /**
-     * __Deprecated - use `<TruncateText /> instead`__
-     */
-    ellipsis: PropTypes.bool
-    /* eslint-enable react/require-default-props */
-  }
+const Heading = (props) => {
+  useDeprecated({
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/index.js:46]: Only after TruncateText is migrated

File: packages/ui-heading/src/Heading/styles.js (hunk 4 above)
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/styles.js:82]: It's up to the linter/prettier

File: packages/ui-heading/src/Heading/styles.js (hunk 5 above)
Comment (HerrTopi) [reply packages/ui-heading/src/Heading/styles.js:101]: It's up to the linter/prettier

File: packages/ui-img/src/Img/__tests__/Img.test.js (hunk 6 above)
Comment (HerrTopi) [reply packages/ui-img/src/Img/__tests__/Img.test.js:66]: It is possible, though not necessary. We have to find a way to unit test the styles.js. These  tests will be covered there. (I'll create a ticket for that)

**Pull request closed without merging**